### PR TITLE
refactor: update npm scripts to use the new dsp

### PR DESCRIPTION
**Pull request by tomivirkki**
This PR makes the dspublisher scripts use the NPM package release of DSP (instead of the Docker image).

Fixes https://github.com/vaadin/docs-app/issues/281
Fixes https://github.com/vaadin/docs-app/issues/331
Fixes https://github.com/vaadin/docs-app/issues/330
Closes https://github.com/vaadin/docs-app/issues/307
Fixes https://github.com/vaadin/docs-app/issues/278
Fixes https://github.com/vaadin/docs-app/issues/269

### CODE REVIEW

File: dspublisher/dspublisher-scripts.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (tomivirkki) [inline dspublisher/dspublisher-scripts.js:1]: This file was added to hide the verbose output that comes from starting `docs` and `docs-app` in parallel in favor of a single progress bar + status message.

**Comment by jouni** [comment]
> It also hides the DSP process output and displays a nice progress bar instead:

I assume warnings and errors will still be logged in addition to this progress bar?

File: dspublisher/dspublisher-scripts.js [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        // The current command was finished, run the next one
+        runCommand(commands, index + 1);
+      } else {
+        // All commands were run, finish
+        finish();
+      }
+    });
+
+    // TODO: --verbose option
+    process.stdout.on('data', (data) => {
```
Comment (tomivirkki) [inline dspublisher/dspublisher-scripts.js:250]: > I assume warnings and errors will still be logged in addition to this progress bar?

Errors do get logged, everything else gets hidden for now. Currently, the script runs a command, then listen to the output stream here, and reacts when the stream contains specific strings.

Comment (jouni) [reply dspublisher/dspublisher-scripts.js:250]: Warnings regarding broken cross references are the most useful at the moment, and I assume our own documentation authors rely on those. Sure, we can still run docs-app manually to see those, but regular users will have a difficult time tracking down those.

Comment (tomivirkki) [reply dspublisher/dspublisher-scripts.js:250]: The intention is to still add the verbose mode as suggested [here](https://github.com/vaadin/docs-app/issues/281#issuecomment-1062818834).

But I agree that the default mode should probably also show warnings.

**Comment by jouni** [comment]
After showing `Ready. Open http://localhost:8000 in the browser.`, we could also show instructions how to stop the process.

File: articles/tools/dspublisher/devmode.asciidoc [hunk 3]
```diff
@@ -5,15 +5,15 @@ order: 30
...
 
 [source,terminal]
 ----
 npm run dspublisher:start
 ----
 
-Note that the first run needs to download the Design System Publisher Docker image and install a lot of (Maven and npm) dependencies so it might take a good while (about ten minutes or so, depending on the network connection speed) to have it fully running.
-Following start-ups are quicker (about four minutes) as the dependencies are already available.
+Note that the first run needs to download the Design System Publisher package and install a lot of (Maven and npm) dependencies so it might take a good while to have it fully running.
+Following start-ups are quicker as the dependencies are already available.
```
Comment (dougjewsbury) [inline articles/tools/dspublisher/devmode.asciidoc:16]:
```suggestion
Subsequent start-ups are quicker, as the dependencies are already available.
```

**Review by jouni — APPROVED** [review]

**Comment by jouni** [comment]
We should cherry-pick this to the `dsp` branch as well, right?

**Comment by tomivirkki** [comment]
> We should cherry-pick this to the dsp branch as well, right?

That's right

**Pull request merged**